src/objects/tests/v2/test_auth_fields.py

```diff
@@ -0,0 +1,375 @@
+            object_type=self.object_type,
+            mode=PermissionModes.read_only,
+            token_auth=self.token_auth,
+            use_fields=True,
+            fields=["url", "type", "record__startAt"],
+        )
+        object = ObjectFactory.create(object_type=self.object_type)
+        record = ObjectRecordFactory.create(object=object, data={"name": "some"})
+        url = reverse("object-detail", args=[object.uuid])
+
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.json(),
+            {
+                "url": f"http://testserver{url}",
+                "type": self.object_type.url,
+                "record": {"startAt": record.start_at.isoformat()},
+            },
+        )
+        self.assertEqual(
+            set(response._headers["x-unauthorized-fields"][1].split(",")),
+            {
+                "uuid",
+                "record__data__name",
+                "record__correctionFor",
+                "record__endAt",
+                "record__correctedBy",
+                "record__registrationAt",
+                "record__index",
+                "record__geometry__coordinates",
+                "record__geometry__type",
+                "record__typeVersion",
+            },
+        )
+
+    def test_retrieve_with_query_fields(self):
+        PermissionFactory.create(
+            object_type=self.object_type,
+            mode=PermissionModes.read_only,
+            token_auth=self.token_auth,
+            use_fields=True,
+            fields=["url", "type", "record"],
+        )
+        object = ObjectFactory.create(object_type=self.object_type)
+        record = ObjectRecordFactory.create(object=object)
+        url = reverse("object-detail", args=[object.uuid])
+
+        response = self.client.get(url, {"fields": "url,type,record__data__name"})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.json(),
+            {
+                "url": f"http://testserver{url}",
+                "type": self.object_type.url,
+                "record": {"data": {"name": record.data["name"]}},
+            },
+        )
+        self.assertNotIn("x-unauthorized-fields", response._headers)
+
+    def test_retrieve_incorrect_auth_fields(self):
+        PermissionFactory.create(
+            object_type=self.object_type,
+            mode=PermissionModes.read_only,
+            token_auth=self.token_auth,
+            use_fields=True,
+            fields=["url", "type", "some"],
+        )
+        object = ObjectFactory.create(object_type=self.object_type)
+        ObjectRecordFactory.create(object=object)
+        url = reverse("object-detail", args=[object.uuid])
+
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
```

Comment:
Good call!

This is probably the best behaviour and should be resolved by the admin.